File: scripts/compare-float-output.ts
import { compareFloatOutputs } from "./lib/float-compare";

function getOption(args: string[], name: string): string | undefined {
  const index = args.indexOf(name);
  return index >= 0 ? args[index + 1] : undefined;
}

function requireOption(args: string[], name: string): string {
  const value = getOption(args, name);
  if (value == null) {
    throw new Error(`Missing option: ${name}`);
  }
  return value;
}

function main() {
  const args = process.argv.slice(2);
  const actualPath = requireOption(args, "--actual");
  const expectedPath = requireOption(args, "--expected");
  const tolerance = Number.parseFloat(requireOption(args, "--tolerance"));

  if (!Number.isFinite(tolerance)) {
    throw new Error("Tolerance must be a finite number");
  }

  const ok = compareFloatOutputs({ actualPath, expectedPath, tolerance });
  process.exit(ok ? 0 : 1);
}

try {
  main();
} catch (error) {
  const message = error instanceof Error ? error.message : String(error);
  console.error(message);
  process.exit(1);
}

File: scripts/compare-float-output.py
#!/usr/bin/env python3
# /// script
# requires-python = ">=3.12"
# ///
"""浮動小数点出力の比較 CLI

Usage:
    python3 scripts/compare-float-output.py --actual FILE --expected FILE --tolerance TOL
    Exit 0 = match, Exit 1 = mismatch
"""
import argparse
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))
from lib.float_compare import compare_float_outputs


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--actual", required=True)
    parser.add_argument("--expected", required=True)
    parser.add_argument("--tolerance", required=True, type=float)
    args = parser.parse_args()

    if not args.tolerance >= 0:
        print("Tolerance must be a non-negative finite number", file=sys.stderr)
        sys.exit(1)

    ok = compare_float_outputs(args.actual, args.expected, args.tolerance)
    sys.exit(0 if ok else 1)


if __name__ == "__main__":
    main()
